### PR TITLE
Reimplementation of Lisa frontend auth calls

### PR DESCRIPTION
**Pull request by chrisjameswright**
- Simplifying AuthorisationService code to remove multiple Auth calls and RuntimeException cases.
- Removal of unneeded UserDetails code and associated tidy-up.

### CODE REVIEW

File: app/controllers/LisaBaseController.scala [hunk 1]
```diff
@@ -48,12 +48,9 @@ trait LisaBaseController extends FrontendController
   }
 
   private def isReapplication(user: UserAuthorised)(implicit request: Request[AnyContent]): Future[Boolean] = {
-    shortLivedCache.fetchAndGetEntry[Boolean](s"${user.internalId}-lisa-registration", Reapplication.cacheKey) map { bool =>
-      bool
-      match {
-        case Some(true) => true
-        case _ => false
-      }
+    shortLivedCache.fetchAndGetEntry[Boolean](s"${user.internalId}-lisa-registration", Reapplication.cacheKey) map {
```
Comment (drewmiley) [inline app/controllers/LisaBaseController.scala]: .getOrElse(false) ?

File: app/services/AuthorisationService.scala [hunk 2]
```diff
@@ -17,86 +17,54 @@
...
-              Logger.info("The enrolment has not been authorised yet so checking Enrolments service the group id is " + groupId)
-              taxEnrolmentService.getNewestLisaSubscription(groupId)(hc) map {
+    ).retrieve(internalId and groupIdentifier and authorisedEnrolments) {
+      case Some(id) ~ Some(groupId) ~ enrolments =>
+        statusFromAuth(id, enrolments)
+          .map(a => Future.successful(Some(a)))
+          .getOrElse(statusFromTaxEnrolments(id, groupId))
+          .map(_.getOrElse(UserAuthorised(id, TaxEnrolmentDoesNotExist)))
+      case _ => Future.successful(UserUnauthorised)
+    } recover {
```
Comment (drewmiley) [inline app/services/AuthorisationService.scala]: Curious, what does recover do?

File: app/services/AuthorisationService.scala [hunk 3]
```diff
@@ -17,86 +17,54 @@
...
-          res match {
-            case Right(zref) => {
-              Logger.info("HMRC-LISA-ORG Enrolment is Authorised")
-              Future.successful(UserAuthorisedAndEnrolled(userId, user, zref))
-            }
-            case Left(_) => {
-
-              Logger.info("The enrolment has not been authorised yet so checking Enrolments service the group id is " + groupId)
-              taxEnrolmentService.getNewestLisaSubscription(groupId)(hc) map {
+    ).retrieve(internalId and groupIdentifier and authorisedEnrolments) {
```
Comment (drewmiley) [inline app/services/AuthorisationService.scala]: Curious, what does retrieve do?

File: app/services/AuthorisationService.scala (hunk 2 above)
Comment (chrisjameswright) [reply app/services/AuthorisationService.scala]: Easiest to think of it as a bit like a `catch` in other languages - it's a partial function for handling Throwables.

File: app/services/AuthorisationService.scala (hunk 3 above)
Comment (chrisjameswright) [reply app/services/AuthorisationService.scala]: Retrieve is used to fetch one or more items out of Auth, assuming the predicates in the authorised section passed.

**Comment by chrisjameswright** [comment]
Will continue LISA-2345 off this branch as the testing required will be the same for both.

**Pull request closed without merging**